### ISSUE TITLE
SAML SSO: automatically fill the Full Name and add UPN field for provisioned users 

### ISSUE THREAD

**Issue by vincentbriglia** [docker/roadmap#310]
**Tell us about your request**
To create a stop-gap measure in absence of the REST API and SAML SCIM, when a user logs in through SAML SSO, it would be great to:
- populate the `Full Name` from the iDP information, which will make it easier to look up a user.
- add the [UPN](https://docs.microsoft.com/en-us/azure/active-directory/hybrid/plan-connect-userprincipalname) to allow associating user back to the AzureAD ([possible solution](https://community.auth0.com/t/add-the-upn-to-event-user/60839))

**Which service(s) is this request for?**
Docker Hub

**Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?**
- [ ] #309 

**Are you currently working around the issue?**
Currently a manual process, even though a user can't log in through SSO, the license is still consumed if the user isn't removed.

**Comment by tehautanop** [comment]
Related to this, as of today it seems DockerHub organisation page now displays also the users email addresses 

We need a way to have the entire organisation users data exported (csv or similar) either from DockerHub organisation page and/or using hub-tool

**Comment by tehautanop** [comment]
https://github.com/docker/roadmap/issues/316

**Comment by ChefAustin** [comment]
> Related to this, as of today it seems DockerHub organisation page now displays also the users email addresses
> 
> We need a way to have the entire organisation users data exported (csv or similar) either from DockerHub organisation page and/or using hub-tool

Agreed. While the webUI/DockerHub console might be displaying email address, the `hub-tool` CLI does not. Would be nice to have this data available via `hub-tool` for programmatic access.
